types/v56/types.go

```diff
@@ -1360,11 +1360,107 @@ type VM struct {
+
+	VmSpecSection *VmSpecSection `xml:"VmSpecSection,omitempty"` // Container for the specification of this virtual machine. This is an alternate to using ovf:VirtualHardwareSection + ovf:OperatingSystemSection
+}
+
+// VmSpecSection from VM struct
+type VmSpecSection struct {
+	Modified          *bool             `xml:"Modified,attr,omitempty"`
+	Info              string            `xml:"ovf:Info"`
+	OsType            string            `xml:"OsType,omitempty"`            // The type of the OS. This parameter may be omitted when using the VmSpec to update the contents of an existing VM.
+	NumCpus           *int              `xml:"NumCpus,omitempty"`           // Number of CPUs. This parameter may be omitted when using the VmSpec to update the contents of an existing VM.
+	NumCoresPerSocket *int              `xml:"NumCoresPerSocket,omitempty"` // Number of cores among which to distribute CPUs in this virtual machine.. This parameter may be omitted when using the VmSpec to update the contents of an existing VM.
+	CpuResourceMhz    *CpuResourceMhz   `xml:"CpuResourceMhz,omitempty"`    // CPU compute resources. This parameter may be omitted when using the VmSpec to update the contents of an existing VM.
+	MemoryResourceMb  *MemoryResourceMb `xml:"MemoryResourceMb"`            // Memory compute resources. This parameter may be omitted when using the VmSpec to update the contents of an existing VM.
+	MediaSection      *MediaSection     `xml:"MediaSection,omitempty"`      // The media devices of this VM.
+	DiskSection       *DiskSection      `xml:"DiskSection,omitempty"`       // virtual disks of this VM.
+	HardwareVersion   *HardwareVersion  `xml:"HardwareVersion"`             // vSphere name of Virtual Hardware Version of this VM. Example: vmx-13 This parameter may be omitted when using the VmSpec to update the contents of an existing VM.
+	VmToolsVersion    string            `xml:"VmToolsVersion,omitempty"`    //	VMware tools version of this VM.
+	VirtualCpuType    string            `xml:"VirtualCpuType,omitempty"`    // The capabilities settings for this VM. This parameter may be omitted when using the VmSpec to update the contents of an existing VM.
+	TimeSyncWithHost  *bool             `xml:"TimeSyncWithHost,omitempty"`  // Synchronize the VM's time with the host.
+}
+
+// DiskSection from VM/VmSpecSection struct
+type DiskSection struct {
+	DiskSettings []*DiskSettings `xml:"DiskSettings"`
+}
+
+// DiskSettings from VM/VmSpecSection/DiskSection struct
+type DiskSettings struct {
+	DiskId              string     `xml:"DiskId,omitempty"`              // Specifies a unique identifier for this disk in the scope of the corresponding VM. This element is optional when creating a VM, but if it is provided it should be unique. This element is mandatory when updating an existing disk.
```

Comment:
Is it unique among all VMs or unique within the VM? Please specify in the comment

Reply:
don't have info for that

Reply:
According info I get, ID are unique within VM.